compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/types/BTupleType.java

```diff
@@ -92,9 +93,11 @@ private BTupleType(BTypeSymbol tsymbol, boolean readonly) {
 
     @Override
     public List<BType> getTupleTypes() {
-        List<BType> types = new ArrayList<>();
-        members.forEach(member -> types.add(member.type));
-        return types;
+        if (memberTypes == null) {
+            memberTypes = new ArrayList<>();
+            members.forEach(member -> memberTypes.add(member.type));
+        }
+        return memberTypes;
     }
 
     public List<BTupleMember> getTupleMembers() {
```

Comment:
Shall we rename this to
```suggestion
    public List<BTupleMember> getMembers() {
```

Reply:
Should we rename  [L94](https://github.com/ballerina-platform/ballerina-lang/pull/39245/files/5402a36f94089e05506275fab1d129af423db2ec#diff-3b226cb8bddf2583036613e2a28068a8868618c91c81984bbfd0f5b5b3d6b3a3L94) as well

Reply:
We could rename since it is internal. I would rename it to `getMemberTypes`, but let's not do that in this PR. Maybe separately.